frame/offences/src/migration.rs

```diff
@@ -0,0 +1,67 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2018-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+mod deprecated {
+    use crate::{Config, Perbill, OffenceDetails, SessionIndex};
+    use frame_support::{decl_module, decl_storage};
+    use sp_std::prelude::*;
+
+    /// Type of data stored as a deferred offence
+    pub type DeferredOffenceOf<T> = (
```

Comment:
why do you need to duplicate this?

Reply:
My initial idea was deprecate this type, but once the `deprecated` module is not public I needed it for `set_deferred_offences` method.
Once the `set_deferred_offences` is not needed anymore, now this type is defined only here.